inject.sh

```diff
@@ -1,135 +1,140 @@
-fi
+    replace_or_insert "SESSION_DRIVER" "$SESSION_DRIVER"
+ fi
 if [ "$SESSION_LIFETIME" != '' ]; then
-    sed -i "s|SESSION_LIFETIME=.*|SESSION_LIFETIME=${SESSION_LIFETIME}|i" /conf/.env
-fi
+    replace_or_insert "SESSION_LIFETIME" "$SESSION_LIFETIME"
+ fi
 if [ "$QUEUE_CONNECTION" != '' ]; then
-    sed -i "s|QUEUE_DRIVER=.*|QUEUE_DRIVER=${QUEUE_DRIVER}|i" /conf/.env
-fi
+    replace_or_insert "QUEUE_CONNECTION" "$QUEUE_CONNECTION"
+ fi
 if [ "$SECURITY_HEADER_HSTS_ENABLE" != '' ]; then
-    sed -i "s|SECURITY_HEADER_HSTS_ENABLE=.*|SECURITY_HEADER_HSTS_ENABLE=${SECURITY_HEADER_HSTS_ENABLE}|i" /conf/.env
-fi
+    replace_or_insert "SECURITY_HEADER_HSTS_ENABLE" "$SECURITY_HEADER_HSTS_ENABLE"
+ fi
 if [ "$SECURITY_HEADER_CSP_CONNECT_SRC" != '' ]; then
-    sed -i "s|SECURITY_HEADER_CSP_CONNECT_SRC=.*|SECURITY_HEADER_CSP_CONNECT_SRC=${SECURITY_HEADER_CSP_CONNECT_SRC}|i" /conf/.env
-fi
+    replace_or_insert "SECURITY_HEADER_CSP_CONNECT_SRC" "$SECURITY_HEADER_CSP_CONNECT_SRC"
+ fi
 if [ "$SECURITY_HEADER_SCRIPT_SRC_ALLOW" != '' ]; then
-    sed -i "s|SECURITY_HEADER_SCRIPT_SRC_ALLOW=.*|SECURITY_HEADER_SCRIPT_SRC_ALLOW=${SECURITY_HEADER_SCRIPT_SRC_ALLOW}|i" /conf/.env
-fi
+    replace_or_insert "SECURITY_HEADER_SCRIPT_SRC_ALLOW" "$SECURITY_HEADER_SCRIPT_SRC_ALLOW"
+ fi
 if [ "$SESSION_SECURE_COOKIE" != '' ]; then
-    sed -i "s|SESSION_SECURE_COOKIE=.*|SESSION_SECURE_COOKIE=${SESSION_SECURE_COOKIE}|i" /conf/.env
-fi
-if [ "$REDIS_SCHEME" != '' ]; then
-    sed -i "s|REDIS_SCHEME=.*|REDIS_SCHEME=${REDIS_SCHEME}|i" /conf/.env
-fi
-if [ "$REDIS_PATH" != '' ]; then
-    sed -i "s|REDIS_PATH=.*|REDIS_PATH=${REDIS_PATH}|i" /conf/.env
-fi
-if [ "$REDIS_HOST" != '' ]; then
-    sed -i "s|REDIS_HOST=.*|REDIS_HOST=${REDIS_HOST}|i" /conf/.env
-fi
-if [ "$REDIS_PORT" != '' ]; then
-    sed -i "s|REDIS_PORT=.*|REDIS_PORT=${REDIS_PORT}|i" /conf/.env
-fi
-if [ "$REDIS_PASSWORD" != '' ]; then
-    sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${REDIS_PASSWORD}|i" /conf/.env
-elif [ "$REDIS_PASSWORD_FILE" != '' ]; then
-    value=$(<$REDIS_PASSWORD_FILE)
-    sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${value}|i" /conf/.env
-fi
+    replace_or_insert "SESSION_SECURE_COOKIE" "$SESSION_SECURE_COOKIE"
+ fi
+# if [ "$REDIS_SCHEME" != '' ]; then
+#     sed -i "s|REDIS_SCHEME=.*|REDIS_SCHEME=${REDIS_SCHEME}|i" /conf/.env
+# fi
+# if [ "$REDIS_PATH" != '' ]; then
+#     sed -i "s|REDIS_PATH=.*|REDIS_PATH=${REDIS_PATH}|i" /conf/.env
+# fi
+# if [ "$REDIS_HOST" != '' ]; then
+#     sed -i "s|REDIS_HOST=.*|REDIS_HOST=${REDIS_HOST}|i" /conf/.env
+# fi
+# if [ "$REDIS_PORT" != '' ]; then
+#     sed -i "s|REDIS_PORT=.*|REDIS_PORT=${REDIS_PORT}|i" /conf/.env
+# fi
+# if [ "$REDIS_PASSWORD" != '' ]; then
+#     sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${REDIS_PASSWORD}|i" /conf/.env
+# elif [ "$REDIS_PASSWORD_FILE" != '' ]; then
+#     value=$(<$REDIS_PASSWORD_FILE)
+#     sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${value}|i" /conf/.env
+# fi
```

Comment:
I'm inclined to keep these while they still exist in Lychee's `.env.example`, but I could be convinced.

Reply:
I would only put them once that they are actually used by Lychee. Not before. :)